### DISCUSSION TITLE
Why is the cherry picking transient prefix called Apply?

### DISCUSSION THREAD

**Discussion by jacobilsoe** [magit/magit#5234] [Support] [answered]
There is probably a good reason but why is the cherry picking transient prefix called `Apply` and not e.g. `Cherry pick`?
```
"A" "Apply"          magit-cherry-pick
```

**Comment by tarsius** [comment, marked as answer]
Because these commands do in fact *apply* commits.

This is how its manpage describes what `git cherry-pick` does:

> git-cherry-pick - Apply the changes introduced by some existing commits

*Cherry picking* is overly specific to the point of often being inaccurate. One might very well use `git cherry-pick` to move some *bad apples* from a feature branch onto a "clean up this crap later" branch.

Also we use `A` as the key binding because applying a commit is very similar to applying just the *changes* in a commit (which is bound to `a`) [without also bringing along the commit message and other metadata from the commit] [^1]. Which should make this more discoverable and easier to remember.

[^1]: Re-reading the `git-cherry-pick`'s manpage after writing this, I cannot help but think that that isn't such a good summary. After all it does actually bring along more than just the *changes*.

**Reply by jacobilsoe** [reply]
Thank you for the detailed answer. 👍
